client/client_test.go

```diff
@@ -311,3 +311,13 @@ func (cs *clientSuite) TestDebugGet(c *C) {
 	c.Check(cs.reqs[0].URL.Path, Equals, "/v1/debug")
 	c.Check(cs.reqs[0].URL.Query(), DeepEquals, url.Values{"action": []string{"do-something"}, "foo": []string{"bar"}})
 }
+
+func (cs *clientSuite) TestNonExistentSocketErrors(c *C) {
+	cli, err := client.New(&client.Config{Socket: "/tmp/not-the-droids-you-are-looking-for-bm90IHRoZSBkcm9pZHMgeW91IGFyZSBsb29raW5nIGZvcgo"})
+	c.Check(err, IsNil)
+
+	_, err = cli.SysInfo()
+	c.Check(err, NotNil)
+	var notFoundErr *client.SocketNotFoundError
+	c.Check(errors.As(err, &notFoundErr), Equals, true)
```

Comment:
Would be good to also test that the `notFoundErr.SocketPath` field is what we expect, and the `notFoundErr.Err` field is not nil.